Add benchmarks and Cargo profile using link-time optimization (#89)
feos-org/feos: README.md

pip install git+https://github.com/feos-org/feos
```

> This command builds the package without link-time optimization (LTO) which results in considerably worse performance.

Reviewer: slightly negative wording imo vielleicht
```suggestion
This command builds the package without link-time optimization (LTO) that can be used to increase the performance further.
```

See the *Building from source* section for information about building the wheel including LTO.

### Building from source

To compile the code you need the Rust compiler and `maturin` (>=0.13,<0.14) installed.
To install the package directly into the active environment (virtualenv or conda), use

```
maturin develop --release
```

which uses the `python` and `all_models` feature as specified in the `pyproject.toml` file.

Alternatively, you can specify the models or features that you want to include in the python package explicitly, e.g.

```
maturin develop --release --features "python pcsaft dft"
```

for the PC-SAFT equation of state and Helmholtz energy functional.

To build wheels including link-time optimization (LTO), use

```
maturin build --profile="release-lto"
```

which will use the `python` and `all_models` features specified in the `pyproject.toml` file.
Use the following command to build a wheel with specific features:

```
maturin build --profile="release-lto" --features "python ..."
```

> LTO increases compile times drastically but the resulting wheel is way more performant and has a smaller size.

Reviewer: maybe:
```suggestion
LTO increases compile times measurably but the resulting wheel is more performant and has a smaller size.
```
